File: Topico_5/intercal.hs
intercal [] y = y
intercal x [] = x
intercal (x:xs) (y:ys) = [x] ++ [y] ++ intercal xs ys

main = do
    a <- readLn :: IO [Int]
    b <- readLn :: IO [Int]
    print $ intercal a b

File: Topico_5/intercal.js
function intercal(list, list2) {
  var arr = [];
  arr = list.map((valor, i) => { 
         arr.push(valor);
         if (list2[i]) {
            arr.push(list2[i]);
         }
         return arr;
  }); 
    return arr[0];
}

console.log(intercal([1,2,3], [4,5,6]));